### ISSUE TITLE
Increasing validation loss in RDSM with time-varying data

### ISSUE THREAD

**Issue by ryuzakizh** [autonlab/auton-survival#91]
Hi auton-survival community @chiragnagpal @Jeanselme @chufangao @salvaRC , I appreciate your contribution to the time-varying survival analysis and thank you for making this library open to public.

I am having an issue with training RDSM using my own custom dataset. I made sure that my dataset looks like the one you used in jupyter notebook tutorials; my `T` column is the remaining time to event, and `E` is an event indicator. However, during training I have been continuously seeing decreasing training loss but increasing validation loss. 

Then I tried to see what happens if I train with  PBC dataset that you used in demo notebooks, and I noticed the same situation there; decreasing training loss and increasing validation loss.

I haven't made any changes to the methodology. Is this something intrinsic to RDSM or am I doing something wrong? These are the logs from the model

![train_val_losses](https://user-images.githubusercontent.com/27948894/189642895-a5522bc9-dc18-4494-b963-73b0ece75817.png)

**Comment by Jeanselme** [comment]
Hello, 
I have just dug into this, I am not able to reproduce the problem. Are you using the RDSM available in `auton_survival/models/dsm`? Do you have this issue when running `examples/RDSM on PBC Dataset.ipynb` ? Feel free to share your code if I can help with this

**Comment by Jeanselme** [comment]
Hello,

I have now tried both codes. I have updated the `Jeanselme/` one as the normalization was not adapted for time series and was raising an error. I still do not observe the issue with the PBC dataset. Can you please check if the problem still occurs for you with the latest version? If so, please share with me (by email) the whole repo and I will dig into this.

**Comment by ryuzakizh** [comment]
Hi @Jeanselme , thanks a lot for the update.

I trained the latest version of RDSM on PBC dataset, and seems like it's fine. Validation loss decreases for the first 60 iterations and then starts to increase, but it stops shortly after that. Perhaps it starts to overfit the data since PBC itself is a small dataset.

I am still getting the issue when I use my own data though. But now I know that it might be a dataset problem, and it's not related to the methodology. Thank you!

**Comment by Jeanselme** [comment]
Hello,

Great to hear you no longer have the issue with PBC. The training should stop after a few iterations of increasing loss on the validation set and return the best model observed. 

Concerning your dataset, are you normalizing the data?

**Comment by ryuzakizh** [comment]
> Hello,
> 
> Great to hear you no longer have the issue with PBC. The training should stop after a few iterations of increasing loss on the validation set and return the best model observed.
> 
> Concerning your dataset, are you normalizing the data?

Thanks for the quick reply, I am normalizing only continuous variables. But I found out that in PBC all variables irrespective of the type are being normalized, so I decided to do the same and see what happens.

**Comment by ryuzakizh** [comment]
Hello @Jeanselme , I am still getting the same issue, I have been digging into this for a week. Could you please possibly look at it with fresh eyes if I send you the repository by email? I would very much appreciate it.

**Comment by chiragnagpal** [comment]
@ryuzakizh @Jeanselme Is this resolved yet? If not can I help facilitate this ?

**Comment by ryuzakizh** [comment]
Hello @chiragnagpal, no the issue hasn't been resolved. This was related to my thesis, so I ended up reporting the results that I got, and I still don't know what the problem was. However, we may consider this problem no longer relevant.
